### ISSUE TITLE
A Suspected Bug in binary_crossentropy

### ISSUE THREAD

**Issue by River861** [keras-team/keras#15249]
Hi, I found that the the calculation result of `binary_crossentropy` loss function is different from other deep learning libraries, such as theano.

Here is an example code using keras2.6 and tensoflow2.6.0:
```python
import numpy as np
import keras

y_true = np.array([0., 1., 0.], dtype=np.float32)
y_pred = np.array([0.9999999, 0.9999999, 0.0000001], dtype=np.float32)

res = keras.backend.binary_crossentropy(y_true, y_pred)

print("loss = ", res.numpy())
```
The result is:
```
loss =  [15.333239 -0.       -0.      ]
```

And this is the code using theano1.0.4:
```python
import numpy as np
from theano import tensor as T

y_true = np.array([0., 1., 0.], dtype=np.float32)
y_pred = np.array([0.9999999, 0.9999999, 0.0000001], dtype=np.float32)

res = T.nnet.binary_crossentropy(y_pred, y_true)

print("loss = ", res.eval())
```
The result is:
```
loss =  [1.5942385e+01 1.1920930e-07 1.1920930e-07]
```

I then found the cause of the inconsistency is that, tensorflow use the `epsilon`  to caculate the loss value,  which I think is redundant because the output has been clipped using `epsilon` eariler. Here is the source code location:
https://github.com/keras-team/keras/blob/372515ca4540d99eece842ab0b56d13eedf37b7c/keras/backend.py#L5164-L5169

Besides, I found that the `categorical_crossentropy` doesn't use the `epsilon` to caculate the loss value. This makes me more suspicious of the usage of `epsilon` in `binary_crossentropy`:
https://github.com/keras-team/keras/blob/372515ca4540d99eece842ab0b56d13eedf37b7c/keras/backend.py#L5023-L5025

I am not sure if I should submit this issue to TensorFlow or Keras, so I submit this issue on both sides. I'm sorry if this caused any trouble.

Thanks.

**Comment by michele-sama-dialpad** [comment]
Hi there, is it there any update on this one?
